packages/grafana-data/src/transformations/transformers/joinByField.ts

```diff
@@ -42,6 +42,7 @@ export const joinByFieldTransformer: SynchronousDataTransformerInfo<JoinByFieldO
         }
         const joined = joinDataFrames({ frames: data, joinBy, mode: options.mode });
         if (joined) {
+          joined.refId = `${DataTransformerID.joinByField}-${data.map((frame) => frame.refId).join('-')}`;
```

Comment:
i wonder if this strategy should be shared for all transformations and applied at a bigher level to whatever each transform returns.

are there any cases when this strategy might need to be different between different transforms? 🤔